src/uct/api/uct.h

```diff
@@ -1078,6 +1078,43 @@ ucs_status_t uct_iface_set_am_tracer(uct_iface_h iface, uct_am_tracer_t tracer,
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief Disable synchronous progress for the interface
+ *
+ * Notify the transport that it should avoid doing anything
+ * during uct_worker_progress() and try to progress asynchronously. 
+ * Thus the polling latency of other transports can be reduced.
+ * 
+ * By default the progress is enabled when interface is created.
+ *
+ * @param [in]  iface    Interface to disable progress.
+ * @param [in]  flags    TODO: separately control tx/rx progress 
+ *
+ * @return error code if the progress can not be disabled
```

Comment:
Potentially this may lead to unexpected deadlocks, even so I understand the motivation behind the API. Maybe the right way to achieve the same goal is to decrease priority and introduce delay timer (or something like this)